worker/Dockerfile

```diff
@@ -1,43 +1,50 @@
 # buildbot/buildbot-worker
 
-# Provides a base Ubuntu (14.04) image with latest buildbot worker installed
-# (based on twisted dockerfile https://github.com/cyli/docker-twisted)
+# please follow docker best practices
+# https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/
 
-FROM        ubuntu:14.04
+# Provides a base Ubuntu (16.04) image with latest buildbot worker installed
+# the worker image is not optimized for size, but rather uses ubuntu for wider package availability
+
+FROM        ubuntu:16.04
 MAINTAINER  Buildbot maintainers
 
 COPY . /usr/src/buildbot-worker
 COPY docker/buildbot.tac /buildbot/buildbot.tac
 
 # Last build date - this can be updated whenever there are security updates so
 # that everything is rebuilt
-ENV         security_updates_as_of 2016-08-23
+ENV         security_updates_as_of 2016-10-07
 
 # This will make apt-get install without question
-ENV         DEBIAN_FRONTEND noninteractive
+ARG         DEBIAN_FRONTEND=noninteractive
 
 # Install security updates and required packages
 RUN         apt-get update && \
             apt-get -y upgrade && \
-            apt-get -y install -q \
-            build-essential \
-            git \
-            subversion \
-            python-dev libffi-dev libssl-dev python-pip python-virtualenv curl && \
-            apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
-
+            apt-get -y install -q  -o APT::Get::Install-Recommends="false"  -o APT::Get::Install-Suggests="false" \
```

Comment:
@tardyp are you sure that `APT::Get::Install-Recommends="false"` is a proper key?

```
# docker run -ti --rm ubuntu:16.04 bash -c 'apt-config dump | grep Recommends'
APT::Install-Recommends "1";
# docker run -ti --rm ubuntu:16.04 bash -c 'apt-config dump | grep Suggests'
APT::Install-Suggests "0";
APT::AutoRemove::SuggestsImportant "false";
```

Looks like should be `APT::Install-Recommends`/`APT::Install-Suggests`. Also `APT::Install-Suggests=false` is not required, since this is a default for a base image.

And instead of `APT::Install-Recommends` it's better to use explicit `--no-install-recommends` `apt-get`option.